### PR TITLE
Amended README,POM and represented all the methods on the tests in XML format

### PR DESCRIPTION
**Pull request by pioushasan2305**

### CODE REVIEW

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java [hunk 1]
```diff
@@ -87,7 +96,7 @@ public static void main(String[] args) throws IOException {
                         try {
                             File testDir = new File(testOutput+name);
                             Scanner testReader = new Scanner(testDir);
-                            File temp = new File("new_csv.txt");
+                            File temp = new File("new_xml.txt");
```
Comment (winglam) [inline dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]: What is this file for? Can we provide a more descriptive name?

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java [hunk 2]
```diff
@@ -65,16 +73,17 @@ public static void main(String[] args) throws IOException {
                         startTest = "startAfter";
                         endTest = "endAfter";
                     }
-                    csvPrinter.printRecord(name, startTest, "", timeStamp, "");
+                    createeElemnthead(doc,test,name,timeStamp);
```
Comment (Talank) [inline dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]:
```suggestion
                    createElemnthead(doc,test,name,timeStamp);
```

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java [hunk 3]
```diff
@@ -65,16 +73,17 @@ public static void main(String[] args) throws IOException {
...
+
                         String subFile = argsList.get(inputTestList) + "sootTimerOutput/";
                         try {
                             File subDir = new File(subFile+name);
                             Scanner subReader = new Scanner(subDir);
                             while(subReader.hasNextLine()){
                                 String fullMethodName = subReader.nextLine();
                                 String methodNameInfo = fullMethodName.split(" >>> ")[1];
-                                csvPrinter.printRecord(name, part, methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
+                                createeElemnt(doc,methods,methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
```
Comment (Talank) [inline dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]:
```suggestion
                                createElemnt(doc,methods,methodNameInfo.split(" : ")[0].trim(), methodNameInfo.split(" : ")[1].trim(), returnFrom);
```

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java [hunk 4]
```diff
@@ -98,7 +107,7 @@ public static void main(String[] args) throws IOException {
                                     String method_name = subReader.nextLine();
                                     String subMethod = method_name.split(" >>> ")[1];
                                     if(subMethod.split(" : ")[0].trim().equals(fullMethodName.trim())){
-                                        csvPrinter.printRecord(name, part, subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
+                                        createeElemnt(doc,methods,subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
```
Comment (Talank) [inline dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]:
```suggestion
                                        createElemnt(doc,methods,subMethod.split(" : ")[0].trim(), subMethod.split(" : ")[1].trim(), returnFrom);
```

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java [hunk 5]
```diff
@@ -130,18 +139,84 @@ public static void main(String[] args) throws IOException {
...
             }
-            csvPrinter.flush();
-        } catch (IOException e){
-            System.err.println(e.getMessage());
-        } finally {
-            fileWriter.close();
-            bufferedWriter.close();
+    }
+
+    private static void createeElemnthead(Document doc,Element test,String test_val,String execution_time_val)
```
Comment (Talank) [inline dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]:
```suggestion
    private static void createElemnthead(Document doc,Element test,String test_val,String execution_time_val)
```
Also can you fix it where else it has been called

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java (hunk 1 above)
Comment (pioushasan2305) [reply dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]: @winglam Its a temporary file to save and later parse through sootTimerOutput data. I will rename this to more descriptive name in next pull request.

File: dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java (hunk 5 above)
Comment (pioushasan2305) [reply dt-impact-tracer/src/main/java/edu/washington/cs/dt/impact/util/RuntimeGenerator.java]: @Talank Noted.

**Comment by pioushasan2305** [comment]
In new commit these progress were done.
1. Changed current xml format (Need a bit more changes for required format)
2. Converted the way serialization and deserialization of XML is done to use Jackson(Done)
3. Captured all methods that are transitively invoked by a test on old version(Done)
4. Put Maven surefire runtime information into csv format(Done)
5. Compare time from surefire reports vs XML files you generated(Done)
@winglam @Talank

**Pull request closed without merging**